src/shared/import-util.c

```diff
@@ -14,53 +14,114 @@
 #include "string-table.h"
 #include "string-util.h"
 
-int import_url_last_component(const char *url, char **ret) {
-        const char *e, *p;
-        char *s;
+static const char *skip_protocol_and_hostname(const char *url) {
+        const char *d;
+        size_t n;
+
+        /* A very very lenient implementation of RFC3986 Section 3.2 */
+
+        /* Find colon separating protocol and hostname */
+        d = strchr(url, ':');
+        if (!d || url == d)
+                return NULL;
+        d++;
```

Comment:
Strictly speaking, RFC 3986 says that the hostname ("authority") may be empty:

   The authority component is preceded by a double slash ("//") and is
   terminated by the next slash ("/"), question mark ("?"), or number
   sign ("#") character, or by the end of the URI.

So this parsers eats too much in some cases, and not enough in others. I'm not sure if it matters for the cases where we are using this... But maybe it'd be better to make the parser follow the RFC if it's not much more work?
```C
d = url + strcspn(url, "0-9A-Za-z+-.");
if (d == url)
         return NULL;
if (*d != ':')
         return NULL;
d++;
if (*d != '/')
          return NULL;
d++;
if (*d != '/')
          return NULL;
return d + strcpsn(d, "/?#");
```
?

Reply:
I wanted to be extra lenient here, i.e. not make restrictions on the charset and stuff. i.e. look for the chars that matter, but ignore all others. 

Your proposed code looks for `://` but many urls don't have them, such as mailto and stuff. i'd be careful with that...

The function makes clear that we skip the hostname, i.e. expect there is one, which is an OK assumption for the protocols we care abou there.